### DISCUSSION TITLE
error | cplayer | The osd-margin-x option must be <= 300: 1101

### DISCUSSION THREAD

**Discussion by wbtcpip2** [mpv-player/mpv#14825] [Q&A] [answered]
when setting osd-margin-x to more than 300 i get the error:

cplayer | The osd-margin-x option must be <= 300: 1101

it seems there is a limit to 300 x and 600 y on the osd

is it so?

**Comment by kasper93** [comment]
> it seems there is a limit to 300 x and 600 y on the osd
> 
> is it so?

Yes. https://github.com/mpv-player/mpv/blob/f6d931301bd226239d95f59e3f18abfeab692ee9/sub/osd.c#L61-L62

**Reply by wbtcpip2** [reply]
thank you for the reply.
Can i ask the reason for having this limits?
having a window of 1920 wdth i was expecting to be able to set the margin_x let's say at 480

**Reply by kasper93** [reply]
No reason, it has been like that since the beginning of those options 80270218cb9bda57afbb739fa22f7eb2f3a556ff Patches welcome if you want to increase the limit.

**Reply by wbtcpip2** [reply]
understood thank you. i'm porting a project from libvlc to libmpv so i'm discovering differences now.
I need to place a simple text (show-text command) on a free x,y position on the screen.
Do you know if here is another command or property that allow me to place the text wherever i want x and y position? (without using the osd-align- options)

**Reply by kasper93** [reply, marked as answer]
You can use ASS tags to move it around, no need to use margin options. See `osd-overlay` if you want to have more control.

**Comment by wbtcpip2** [comment]
thank you
